designer/src/fields.tsx

```diff
@@ -441,6 +441,23 @@ const fields: {[key: string]: FieldType} = {
     },
     validationSchema: [['yup.object'], ['yup.nullable']],
   },
+  FAIMSTextField: {
+    'component-namespace': 'faims-custom',
+    'component-name': 'FAIMSTextField',
+    'type-returned': 'faims-core::String',
+    'component-parameters': {
```

Comment:
So I'd suggest we simplify the props here to include just those that specify the field content. Remove eg. 'variant' and 'InputProps' as options, those should be decided by our internal stylesheets.  Then also add a Number field that will render as a TextField with number input (not sure if we need that for BSS).